### PR TITLE
docs,man: Update docs and man pages

### PR DESCRIPTION
**Pull request by k-takata**
* Fix markup of backslash.
* Write about TAG_OUTPUT_FILESEP ptag
* Add more description about TAG_OUTPUT_MODE
* Write about --list-subparsers
* Fix typos.

### CODE REVIEW

File: docs/news.rst [hunk 1]
```diff
@@ -1263,6 +1263,16 @@ The third line means ``\\`` is used when for combining a namespace item
...
+``TAG_OUTPUT_FILESEP``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This pseudo tag represents the separator used in file name: slash or
+backslash.  This is always 'slash' on Unix-like environments.
+This is also 'slash' by default on Windows, however when
+``--output-format=e-tags`` or ``--use-slash-as-filename-separator=no``
+is specified, it becomes 'backslash'.
+
+
```
Comment (masatake) [inline docs/news.rst]: THANK YOU VERY MUCH. This one is I had to do.

File: man/ctags.1.rst.in [hunk 2]
```diff
@@ -1359,6 +1359,10 @@ are not listed here. They are experimental or for debugging purpose.
 	DESCRIPTION
 		Human readable description for the role.
 
+``--list-subparsers[=baselang|all]``
+	Lists the subparsers for a base language for either the specified
+	*baselang* or **all** languages, and then exits.
+
```
Comment (masatake) [inline man/ctags.1.rst.in]: Could you add something like

> **all** is used as default value if the option arugment is omitted.

?

Comment (k-takata) [reply man/ctags.1.rst.in]: Maybe, we should add the same sentence also for other `--list-xxx[=language|all]` options.

Comment (masatake) [reply man/ctags.1.rst.in]: That's right. I will do.

Comment (k-takata) [reply man/ctags.1.rst.in]: I'm preparing it in this PR.

Comment (masatake) [reply man/ctags.1.rst.in]: Oh, thank you.

Comment (k-takata) [reply man/ctags.1.rst.in]: done

File: docs/man/ctags.1.rst [hunk 3]
```diff
@@ -1334,9 +1344,10 @@ are not listed here. They are experimental or for debugging purpose.
 	See :ref:`ctags-optlib(7) <ctags-optlib(7)>`.
 
 ``--list-roles[=language|all[.kinds]]``
-	List the roles for either the specified language or "all"
-	languages. If the parameter kinds is given after the parameter
-	language or "all" with concatenating with '.', list only roles
+	List the roles for either the specified *language* or **all** languages.
```
Comment (masatake) [inline docs/man/ctags.1.rst]: MY TODO: Multiple kinds cannot be specified. s/kinds/a kind/ is needed.

**Pull request merged**